.github/workflows/ci.yml

```diff
@@ -115,48 +115,25 @@ jobs:
             exit 1
           fi
 
-        ## TODO: Replace with `yarn fix:yarn-dedupe` after upgrading to Yarn v3+
-        ## https://yarnpkg.com/cli/dedupe
-        ## https://github.com/yarnpkg/berry/issues/2297
-      - name: Run yarn lint:yarn-deduplicate
+      - name: Run yarn lint:yarn-dedupe
         if: ${{ success() || failure() }}
         run: |
-          if ! yarn lint:yarn-deduplicate; then
+          if ! yarn lint:yarn-dedupe; then
             echo ''
             echo ''
             echo 'ℹ️ ℹ️ ℹ️'
             echo 'Some dependencies can be deduplicated, which will make node_modules'
             echo 'lighter and potentially save us from unexplainable bugs.'
-            echo 'Please run `yarn fix:yarn-deduplicate` locally and commit yarn.lock.'
-            echo 'You may need to run the command 2-3 times in some rare cases.'
+            echo 'Please run `yarn fix:yarn-dedupe` locally and commit yarn.lock.'
             echo 'ℹ️ ℹ️ ℹ️'
             exit 1
           fi
 
-        ## yarn --frozen-lockfile does not work for monorepos, so using a workaround:
-        ## https://github.com/yarnpkg/yarn/issues/5840#issuecomment-467516207
-        ## TODO: Use `yarn install --immutable` after upgrading to Yarn v3+
-      - name: Check yarn.lock stability
-        if: ${{ success() || failure() }}
-        run: |
-          git diff yarn.lock
-          if ! git diff --exit-code yarn.lock; then
-            echo ''
-            echo ''
-            echo 'ℹ️ ℹ️ ℹ️'
-            echo 'Changes were detected in yarn.lock file after running `yarn install`.'
-            echo 'This makes runtime less stable, so should be avoided.'
-            echo 'Please run `yarn install` locally and commit yarn.lock.'
-            echo 'You may also want to run `yarn fix:yarn-deduplicate` just in case.'
-            echo 'ℹ️ ℹ️ ℹ️'
-            exit 1;
-          fi
-
   packages:
     name: NPM Packages
     if: ${{ github.event_name == 'pull_request' }}
     runs-on: ubuntu-20.04
-    timeout-minutes: 30 ## https://github.com/actions/upload-artifact/issues/358
+    timeout-minutes: 60
```

Comment:
I've increased timeout because cold yarn + rust warmup has been causing CI failures: https://github.com/blockprotocol/blockprotocol/actions/runs/3887049425/jobs/6632856500

We are no longer affected by https://github.com/actions/upload-artifact/issues/358 once https://github.com/blockprotocol/blockprotocol/pull/769 has been merged. I've kept a large timeout just in case, but I don't think we need it except for some really rare edge cases.